### PR TITLE
Handle preflight options calls

### PR DESCRIPTION
**Pull request by justinloyola**

### CODE REVIEW

File: h2o-core/src/main/java/water/api/RequestServer.java [hunk 1]
```diff
@@ -222,6 +222,13 @@ public static Route registerEndpoint(String method_uri, Class<? extends RestApiH
   @Override protected void doPost(HttpServletRequest rq, HttpServletResponse rs)   { doGeneric("POST", rq, rs); }
   @Override protected void doHead(HttpServletRequest rq, HttpServletResponse rs)   { doGeneric("HEAD", rq, rs); }
   @Override protected void doDelete(HttpServletRequest rq, HttpServletResponse rs) { doGeneric("DELETE", rq, rs); }
+  @Override protected void doOptions(HttpServletRequest rq, HttpServletResponse rs) {
+    if (System.getProperty(H2O.OptArgs.SYSTEM_PROP_PREFIX + "debug.cors") != null) {
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/api/RequestServer.java]: Minor: DRY - you can use `private static String SYS_PROP_DEBUG_CORS = H2O.OptArgs.SYSTEM_PROP_PREFIX + "debug.cors"`

**Comment by micahstubbs** [comment]
@mmalohlava @justinloyola what do we need to get this merged? it's been 6 days with no activity 🕙

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by micahstubbs** [comment]
👍

**Pull request merged**